core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -151,6 +155,7 @@ public FuzzyQueryBuilder(String fieldName, boolean value) {
      * @param value The value of the term
      */
     public FuzzyQueryBuilder(String fieldName, Object value) {
+        DEPRECATION_LOGGER.deprecated("{} query is deprecated. Instead use the [match] query with fuzziness parameter", NAME);
```

Comment:
> Do you know if there's a way of checking this in the regular REST (yaml) tests?

You can look at the code generated by the docs tests and do what it does. You should add a `feature` section to the test so the clients don't choke on it.

Personally I'm fine with just asserting it in the docs tests. The build still fails if we remove it or change the message in some crazy way and that is all I'm looking for.